types/algoliasearch/index.d.ts

```diff
@@ -1524,6 +1529,31 @@ declare namespace algoliasearch {
     objectID?: string;
```

Comment:
where is the regular Task still used? This can probably be removed for another one

Reply:
There are many instances where it is still used:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f0734581835aa51e2ac5b2edd49788ad3b49169e/types/algoliasearch/index.d.ts#L176
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f0734581835aa51e2ac5b2edd49788ad3b49169e/types/algoliasearch/index.d.ts#L276
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f0734581835aa51e2ac5b2edd49788ad3b49169e/types/algoliasearch/index.d.ts#L444

Reply:
could we consolidate on those, it's fairly inconsistent now

Reply:
I was going to go through all of them, but after finding errors in return types in the Algolia docs it became too time consuming to verify every endpoint. I believe there are about 60 types to update here.

Reply:
Yes, there's a lot to cover...
@squirly are you still thinking on going through all of them...?
I might make time on a couple of hours to do the same.

Reply:
I do not have time this week.

If there are any blocking changes for merging this PR let me know so I can fix them.

Reply:
There are. The [`deleteApiKey`](https://www.algolia.com/doc/api-reference/api-methods/delete-api-key/?language=javascript#response) method for example differs from the regular `Task`.

You've done a lot of the work. But I'd like to go through all of them. I think I can help a bit.

Also, I don't want to be insistent and block your job. If @Haroenv or another contributor agrees, we can merge your PR and work on a more complete one latter.

Reply:
This is a pre-existing issue, not introduced by my changes. So I am unsure why they are blocking. I believe we can make these changes in an iterative way.